aten/src/ATen/Context.h

```diff
@@ -48,13 +48,17 @@ class CAFFE2_API Context {
       LegacyTypeDispatch::TypeUniquePtr{t, LegacyTypeDeleter([](Type* p) { delete p; }) });
   }
 
-  Generator & defaultGenerator(DeviceType device_type) {
+  /* Generator */
+  Generator& getDefaultGenerator(DeviceType device_type, int64_t device = -1) {
```

Comment:
This can take a `Device` instead, which is exactly what these two parameters are.